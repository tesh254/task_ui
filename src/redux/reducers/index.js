import { combineReducers } from "redux";
import loginReducer from "./login";
import taskReducer from "./tasks"

const rootReducer = combineReducers({
  login: loginReducer,
  task: taskReducer
});

export default rootReducer;
