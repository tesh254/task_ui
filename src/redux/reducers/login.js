import { LOGIN_SUCCESS, ERROR, LOADING, AUTH } from "../type";

export let initialState = {
  message: "",
  token: "",
  isAuthenticated: false,
  isLoading: false
};

const loginReducer = (state = initialState, action) => {
  switch (action.type) {
    case LOADING:
      return {
        ...state,
        isLoading: true
      };
    case AUTH:
      return {
        isAuthenticated: action.payload.isAuthenticated
      };
    case LOGIN_SUCCESS:
      let token;

      token = action.payload.accessToken;
      localStorage.setItem("jwt_token", token);
      return {
        ...state,
        isAuthenticated: true,
        isLoading: false,
        token
      };
    case ERROR:
      return {
        ...state,
        isLoading: false
      };
    default:
      return state;
  }
};

export default loginReducer;
