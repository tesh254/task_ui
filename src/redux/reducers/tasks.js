import { FETCH_TASKS, LOADING, ERROR } from "../type";

export let initialState = {
  tasks: [],
  isLoading: true,
  page: 1,
  perPage: 5,
  totalTasks: 5
};

const taskReducer = (state = initialState, action) => {
  switch (action.type) {
    case LOADING:
      return {
        ...state,
        isLoading: true
      };
    case FETCH_TASKS:
        console.log(action.payload.totalTasks / action.payload.perPage)
      return {
        ...state,
        tasks: action.payload.tasks,
        isLoading: false,
        page: action.payload.page,
        lastPage: action.payload.totalTasks / action.payload.perPage,
        perPage: action.payload.perPage,
        totalTasks: action.payload.totalTasks
      };
    case ERROR:
      return {
        isLoading: false
      };
    default:
      return state;
  }
};

export default taskReducer;
