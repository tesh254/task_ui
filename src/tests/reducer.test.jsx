import reducer from "../redux/reducers";
import { initialState } from "../redux/reducers/index";

describe("reducers", () => {
  it("should return the initial state", () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  it("should handle LOADING", () => {
    expect(reducer(initialState, { type: "LOADING" })).toMatchSnapshot();
  });

  it("should handle ERROR", () => {
    expect(reducer(initialState, { type: "ERROR" })).toMatchSnapshot();
  });

  it("should handle LOGIN success", () => {
    expect(
      reducer(initialState, {
        type: "LOGIN_SUCCESS",
        payload: {
          access_token:
            "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjM3OTU5OTI2NDcyLCJkYXRhIjoiMDcwNTE4MTcwNyIsImlhdCI6MTU4MTY2MDAwM30.lUNx_3MCdDxzLvXNyPBjZ9nxe09J62F5M8byF7wvPMg"
        }
      })
    ).toMatchSnapshot();
  });

  it("should handle AUTH type", () => {
      expect(reducer(initialState, {
          type: "AUTH",
          payload: {
              isAuthenticated: true
          }
      })).toMatchSnapshot()
  })
});
