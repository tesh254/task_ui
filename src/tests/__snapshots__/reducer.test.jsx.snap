// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers should handle AUTH type 1`] = `
Object {
  "login": Object {
    "isAuthenticated": true,
  },
}
`;

exports[`reducers should handle ERROR 1`] = `
Object {
  "login": Object {
    "isAuthenticated": false,
    "isLoading": false,
    "message": "",
    "token": "",
  },
}
`;

exports[`reducers should handle LOADING 1`] = `
Object {
  "login": Object {
    "isAuthenticated": false,
    "isLoading": true,
    "message": "",
    "token": "",
  },
}
`;

exports[`reducers should handle LOGIN success 1`] = `
Object {
  "login": Object {
    "isAuthenticated": true,
    "isLoading": false,
    "message": "",
    "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjM3OTU5OTI2NDcyLCJkYXRhIjoiMDcwNTE4MTcwNyIsImlhdCI6MTU4MTY2MDAwM30.lUNx_3MCdDxzLvXNyPBjZ9nxe09J62F5M8byF7wvPMg",
  },
}
`;

exports[`reducers should return the initial state 1`] = `
Object {
  "login": Object {
    "isAuthenticated": false,
    "isLoading": false,
    "message": "",
    "token": "",
  },
}
`;
